src/GpxDocument.cpp

```diff
@@ -558,15 +558,15 @@ QMap<QString, QString> OsmAnd::GpxDocument::readTextMap(QXmlStreamReader& xmlRea
         if (tok == QXmlStreamReader::TokenType::EndElement)
         {
             QString tag = xmlReader.name().toString();
-            if (text != nullptr && text.trimmed().length() > 0)
+            if (!text.isNull() && !text.trimmed().isEmpty())
                 result.insert(tag, text);
-            if (tag.compare(key))
+            if (tag == key)
```

Comment:
Wrong, you should still use ```compare()``` but in a correct way (hint: the return value of this method is int, hint2: see how this method is used throughout the code)